libsmtutil/Z3Interface.cpp

```diff
@@ -169,6 +169,12 @@ z3::expr Z3Interface::toZ3Expr(Expression const& _expr)
 				smtAssert(sortSort, "");
 				return m_context.constant(n.c_str(), z3Sort(*sortSort->inner));
 			}
+			else if (n == "tuple_constructor")
+			{
+				auto constructor = z3::func_decl(m_context, Z3_get_tuple_sort_mk_decl(m_context, z3Sort(*_expr.sort)));
+				smtAssert(constructor.arity() == arguments.size(), "");
+				return constructor();
+			}
```

Comment:
The fix looks quite generic. Why were only empty tuples affected?

I checked a simple case like this and it seems to work fine, but is it possible that it was actually broken in more complex cases of non-empty tuples?
```solidity
contract C {
    function test() public {
        true ? (1, 2) : (3, 4);
    }
}
```

Reply:
That's unlikely. The reason for this bug is that we  forgot to handle an empty tuple in a code that translates our internal classes to z3 expression.

Reply:
So are empty tuples encoded differently than non-empty ones?

Reply:
No, the representation is the same. The problem is that before we get to handling tuples (`tuple_constructor)` there is this check: https://github.com/ethereum/solidity/blob/37d45336f795e1aa5b1d842a98bb29eb63a3c778/libsmtutil/Z3Interface.cpp#L160

So there is a special case for handling expressions with zero arguments, but it did not cover empty tuples before.